### DISCUSSION TITLE
exception_set_exclusive_handler & ISRs

### DISCUSSION THREAD

**Discussion by nstansbury** [earlephilhower/arduino-pico#1821] [General]
I have an initial Platform.IO library of a multi-core "micro-kernel" for the Pico - [Postman::Pico Task Manager](https://github.com/Tactory/Postman) based on the official SDK (`v1.4`) that I am trying to port over to work with the `arduino-pico` core, but I'm not able to set the Systick and SVC ISRs via `exception_set_exclusive_handler(SYSTICK_EXCEPTION, ..)` and `exception_set_exclusive_handler(SVCALL_EXCEPTION, ..)` as they have both already been installed.

Reading some other threads: [ThreadX Porting](https://github.com/earlephilhower/arduino-pico/discussions/292) they seems to suggest this maybe something to do with how the Pico SDK is included in `arduino-pico` and `__isFreeRTOS`.

Could I get some pointers on what might be happening, as I'd just assumed I'd have full access to H/W via the SDK but this appears not to be the case.

**Comment by earlephilhower** [comment]
The SYSTICK is used to handle cycle counting in the plain core.  FreeRTOS will use it for its task swapping.  You could hack and remove cycle count support in non-freertos mode, I suppose.  See `cores/rp2040/RP2040Support.h`
